src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx

```diff
@@ -298,9 +304,12 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
 
   renderRightToolbar = () => {
     const canDelete = this.state.istioObjectDetails !== undefined && this.state.istioObjectDetails.permissions.delete;
+    const istioObject = this.getIstioObject();
+
     return (
       <span style={{ float: 'right' }}>
         <IstioActionDropdown
+          objectKind={istioObject && 'kind' in istioObject ? istioObject.kind : undefined}
```

Comment:
After reviewing this line, I realized that all the Istio Objects should follow the interface IstioObject. Subsequently, all of them should have the kind field. As a result, I saw that we had two deprecated Istio Objects that didn't follow that inheritance: DestinationPolicies and Route Rules.

I created the following PR for removing them: https://github.com/kiali/kiali-ui/pull/1148
Right after this is merged, we will have that all the Istio Objects will be type of IstioObject and then, this check could be delegated to typescript in form of strong type detection.

Reply:
Rebased this PR to use the latest changes.